src/login/logind-session.c

```diff
@@ -1232,6 +1234,10 @@ bool session_may_gc(Session *s, bool drop_not_started) {
                         return false;
         }
 
+        if (s->leader_pidfd >= 0)
+                if (pidfd_is_alive(s->leader_pidfd))
```

Comment:
in new code we really try to at least debug log about unexpected errors, hence maybe:

```c
if (s->leader_pidfd >= 0) {
        r = pidfd_is_alive(s->leader_pidfd);
        if (r < 0)
                log_debug_errno(r, "Unable to determine if leader PID " PID_FMT " is still alive, assuming not.");
        if (r > 0)
                return false;
```